### PR TITLE
Added new messages indicator (#349)

### PR DESCRIPTION
**Pull request by nashvail**
For issue #349 
When user has scrolled beyond a threshold value and new message arrives an indicator is shown. The indicator vanishes when view is scrolled to the bottom.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@nashvail, it looks like you've already signed the Dropbox CLA. Thanks!

File: jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (borisyankov) [inline jsconfig.json]: It is fine to add a VS.code specific file, but maybe do it in a different PR?
Also, if there are other config options needed, feel free to add them to the file.

File: src/common/InfoLabel.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
```
Comment (borisyankov) [inline src/common/InfoLabel.js]: We have an OfflineNotice component that does a very similar thing. I guess we either want to merge this one with it, or make this a more specific one like 'UnreadMsgNotice'.

**Comment by borisyankov** [comment]
We would want to add the text for unread messages in the Chat.js file.
See how OfflineNotice is there.

Also, maybe we want to have one at the top and one at the bottom, depending on wether there are unread messages under and over the current view.

File: src/message-list/MessageList.js [hunk 3]
```diff
@@ -16,17 +19,36 @@ const styles = StyleSheet.create({
...
     };
+
+    // Variables concerned with calculating scroll position
+    this.initialContentHeight = 0;
+    this.initialScrollOffset = 0;
   }
 
   componentWillReceiveProps(nextProps) {
+    // New messages have been received and user has scrolled above the threshold value
+    if (this.hasScrolledBeyondInfoThreshold() && this.didReceiveNewMessages(nextProps)) {
```
Comment (borisyankov) [inline src/message-list/MessageList.js]: You don't need to set any state to the component.
Calculate 'shouldShowNewMessageIndicator' in the render function.
Considering my other note about Chat.js, and moving the notification component there, it should be in the render() of Chat.

File: src/message-list/MessageList.js [hunk 4]
```diff
@@ -45,6 +81,27 @@ export default class MessageList extends React.PureComponent {
...
 
+  // Checks if the current scroll position is beyond threshold scroll position
+  // for showing new messages indicator
+  hasScrolledBeyondInfoThreshold() {
+    return this.scrollOffset > NEW_MSG_INFO_SCROLL_THRESHOLD;
+  }
+
+  // Checks if new messages have been received
+  didReceiveNewMessages(props) {
+    return props.messages.length > this.props.messages.length;
```
Comment (borisyankov) [inline src/message-list/MessageList.js]: This is not sufficient check, I will write some code to do the checks (we currently don't have any)

File: src/common/InfoLabel.js (hunk 2 above)
Comment (nashvail) [inline src/common/InfoLabel.js]: Is it possible to have generalized components like Bootstrap has for such cases? For example a <WarningNotice /> component that can be supplied a text prop to show Offline notice, or any other warning messages. A <InformationNotice /> component to show general information like "Unread messages". Bootstrap has generalized warning, danger, info e.t.c

Comment (borisyankov) [inline src/common/InfoLabel.js]: Yeah, certainly.
Just keep in mind that we should balance between reusability and 'monster component' that tries to do too much stuff.
In that case, I think it is a good idea to have one component that does both.

**Comment by borisyankov** [comment]
I am writing several functions to help with the 'unread messages check'.
Will push soon, so we can move this PR forward.

**Comment by nashvail** [comment]
I guess #486 is the PR that does so ...if I am not wrong. I think the indicator should animate in and also be dismissable, by tapping an x icon located inside the indicator itself. Adding these to todos 👍

**Comment by borisyankov** [comment]
Yeah, that is the PR that will give the unread counts.

@timabbott Is there a possibility a person can have unread messages above current view?
If yes, I guess we should show an unread notice above and below the list accordingly.

**Comment by borisyankov** [comment]
@nashvail I think this PR is the most beneficial to get finished first.
It will also help us see if there are any bugs in the unread count tracking.
When you have time, please focus on it? 💯

**Comment by nashvail** [comment]
Sure @borisyankov! my finals start next week, I am going to try and make some significant progress by this weekend.

**Comment by nashvail** [comment]
Closing this because outdated. New over here -> #537

**Pull request closed without merging**